src/Docfx.Dotnet/DotnetApiCatalog.Compile.cs

```diff
@@ -132,6 +133,11 @@ await LoadCompilationFromProject(project.AbsolutePath) is { } compilation)
                     await Process.Start("dotnet", $"restore \"{path}\"").WaitForExitAsync();
                 }
                 project = await workspace.OpenProjectAsync(path, msbuildLogger);
+
+                foreach (var unresolvedAnalyzer in project.AnalyzerReferences.OfType<UnresolvedAnalyzerReference>())
+                {
+                    Logger.LogWarning("There is .NET Analyzer that can't be resolved. Path: " + unresolvedAnalyzer.FullPath);
```

Comment:
shall we hint user to try explicitly run a dotnet build before running docfx?

Reply:
> shall we hint user to try explicitly run a dotnet build before running docfx?

How about the following text.
> There is .NET Analyzer that can't be resolved. If this analyzer DLL is .NET Source Generator project. Try build with `dotnet build -c Release` before running docfx command.  Path: ";

Reply:
OR if you have a Q&A section in the documentation, it can be referenced. 
Since you are running the `docfx metadata`, I assume you also already have information on how to run `dotnet build` on multi-projects defined in the `docfx.json`.

Reply:
@filzrev Looks good.

@paulushub
> Since you are running the docfx metadata, I assume you also already have information on how to run dotnet build on multi-projects defined in the docfx.json.
It might be problematic to load a project in memory to detect if `dotnet build` is needed, run `dotnet build` and then reload the project to memory. It seems a hint is good enough for now. We can revisit if this issue become more widely spread.